### ISSUE TITLE
New bug in auto indenting

### ISSUE THREAD

**Issue by Muzza** [slynch8/10x#28]
See image. With your cursor at the end of the if( 2 ) line, press return then type {
10x moves the indent backwards so that it aligns with the if( 1 ) rather than if( 2 )

![2021-01-12 08_54_28-AllProjects_win sln - 10x](https://user-images.githubusercontent.com/1583441/104242820-d0916200-54b3-11eb-8cd7-eb3f4ca6022d.png)

**Comment by slynch8** [comment]
doesn't happen for me. Are you using tabs or spaces?

**Comment by Muzza** [comment]
Tabs

**Comment by slynch8** [comment]
is this function a global, and does it happen if it's the first thing in the file? And do you have any editing settings changed? Also, check you're not mixing spaces and tabs in the code

**Comment by Muzza** [comment]
It happens in an empty project with a single empty c file that has nothing in it other than the above.
I've using v0.20.10.0
Perhaps some other setting I'm using is affecting it?

**Comment by Muzza** [comment]
Oh, you mentioned the setting already. I'm not mixing spaces/tabs. (although I could check easier if you implemented viewing whitespace :p )

**Comment by slynch8** [comment]
yeah, too busy breaking stuff to add that feature :-) would you mind sharing your settings file?

**Comment by Muzza** [comment]
```
# The main UI font for 10X. Use the name of the .ttf file in c:\windows\fonts\ (without the extension)
UI.Font:                                          LiberationSans-Regular

# The main UI font size for 10X. It's usually better to change the UI.Scale which also scales the font size.
UI.Font Size:                                     9

# Use 'Horizontal' for standard sub-pixel rendering, 'None' to disable.
# Use VerticalLeft if you have your monitor orientated vertically on its
# left side, and VerticalRight if orientated on the right side.
# Valid values: None, Horizontal, VerticalLeft, VerticalRight
UI.SubPixel Font Rendering:                       Horizontal

# Gamma value for font rendering
UI.Font Gamma:                                    1.8

# The main UI Scale. Set to 1 for 100% scaling, 1.5 for 150% etc. Set to 'auto' to use the OS scale.
# Valid values: float, auto
UI.Scale:                                         auto

# The font for all text editor windows. Use the name of the .ttf file in c:\windows\fonts\ (without the extension)
Editor.Font:                                      JetBrainsMono-Medium

# The font size for all text editor windows
Editor.Font Size:                                 11

# Hide the main menu when not in use. Mouse Over to make visible.
UI.HideMenu:                                      false

# The current colour scheme for all UI elements. Can be 'default'
UI.Colour Scheme:                                 VS+VAX

# How often to backup unsaved files (ms)
Editor.Backup Frequency:                          5000

# Default scripts that 10x has installed to the appdata\10x\PythonScripts path
Python.Installed Scripts:                         Example.py,Snippets.py,RunDebugger.py,FormatFile.py

# Whether the tab key inserts a tab char or n spaces.
# Valid values: Tabs, Spaces
Editor.Tab Mode:                                  Tabs

# Width of a tab in chars
Editor.Tab Width:                                 4

# Path of the Visual Studio vcvars batch file use for building solutions. Restart 10X after changing this value. Set to 'auto' to let 10x try and find the path
Build.Visual Studio VC Vars:                      C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Auxiliary/Build/vcvars64.bat

# Enable VSync
UI.VSync:                                         true

# Scroll bar mode for code text editor panels
# Valid values: Standard, Map
UI.ScrollBarMode:                                 Standard

# Crash dump mode. Restart 10x for this to take effect
# Valid values: Mini, MiniWithHeap, Full
Misc.CrashDumpMode:                               MiniWithHeap

# Number of lines scroll wheel scrolls
UI.MouseWheelScrollStep:                          3

# Maximum time in ms for two mouse clicks to be considered a double click
UI.MouseDoubleClickTime:                          500

# Control+click in text editor to go to definition of symbol. Otherwise just selects word.
UI.ControlClickGotoDefinition:                    true

# Setting to true sleeps the app when idle instead of calling Present. The app is considered idle when nothing on screen changes and there is no user input. Calling nothing but present still uses a small <1% of CPU time, setting this will get it to 0%. However, this can cause issues with G-Sync. A better solution to the G-Sync problem is to turn off G-Sync for 10x in the NVIDIA display settings.
UI.DisablePresentWhenIdle:                        false

# Sends data to PureDev Software, such as time used per day and crashes per day. The data is anonymous. Restart 10x for this setting to take effect.
EnableTelemetry:                                  true

# File wildcards for files that should automatically open in column 0. Separated by commas.
UI.Column0Files:                                  

# File wildcards for files that should automatically open in column 1. Separated by commas.
UI.Column1Files:                                  

# File wildcards for files that should automatically open in column 2. Separated by commas.
UI.Column2Files:                                  

# Show line numbers (true,false)
Editor.Line Numbers:                              true

# Margin width
Editor.Margin Width:                              16

# Delay before the drag-drop dock target panel appears (in ms)
UI.DragDropDockDelay:                             200

# Format lines on paste
Editor.FormatOnPaste:                             false

# Auto-indent new lines
Editor.AutoIndent:                                true

# Highlight all instances of the current symbol
Editor.HighlightCurrentSymbol:                    false

# Highlight all instances of the current word
Editor.HighlightCurrentWord:                      true

# Highlight all instances of the currently selected word
Editor.HighlightSelectedWord:                     true

# Ctrl+tab switch file mode. Can be set to all files or restricted to files open in the current panel
# Valid values: All, Panel
UI.SwitchFileMode:                                Panel

# true to make Toggle Comment command use // line comments instead of /* */
UI.CommentBlockUsesLineComments:                  false

# When creating a new line add tabs or spaces to the current indentation depth
Editor.ExtendEmptyLines:                          false

# The width of the map scroll bar
UI.MapScrollBarWidth:                             60

# Typing an open bracket/brace char ((,{,[) with selected text will enclose the selection with that bracket/brace 
UI.BracketSelection:                              true

# Open the last workspace opened when starting 10x
OpenLastWorkspaceOnStartup:                       false

# True to scroll at 60fps
Editor.Fast Scroll:                               true

# The key repeat rate increases the longer the key is held down. Set to zero to disable.
Editor.Key Repeat Acceleration:                   0

# Time in ms after key is held before it repeats
UI.InitialKeyRepeatDelay:                         500

# Time in ms between key repeats. Set to multiple of 16.6666ms for smooth repeats at 60fps
UI.KeyRepeatDelay:                                33.333333

# Format the current block when a close brace is typed
Parser.FormatOnCloseBrace:                        false

# Path mode for the Search panel results
# Valid values: Full, Relative, Filename, FolderAndFile
Search.PathMode:                                  FolderAndFile

# Search will search files with these extensions
Search.Extensions:                                .asm, .bat, .c, .cc, .cpp, .cs, .css, .csv, .cxx, .def, .disco, .dtd, .fx, .h, .hh, .hpp, .hlsl, .htm, .html, .hxx, .idl, .inc, .inl, .java, .js, .l, .ll, .lnx, .log, .mk, .pl, .pmk, .ps, .py, .s, .sh, .src, .srf, .tlh, .tli, .txt, .vb, .vsh, .wsdl, .xaml, .xml, .xsd, .xsl, .xslt, .y, .psh, .fsh, .sh, .shader, .vsct, .resx, .wxs, .php, .vsixmanifest, .uc, .sql, .webconfig, .aspx, .lua, .cshtml .mm .m .as

# Comma separated list of wildcards for files to exclude
Search.ExcludeList:                               

# The maximum file size that Search will index
Search.MaxFileSize:                               1048576

# Use the current word as the initial search text when opening the Search panel
Search.SerchCurrentWord:                          true

# Spaces are treated as * wildcards when matching files
Search.SpacesAsWildcardsForFindFile:              true

# Search will stop searching after this many results have been found
Search.MaxResults:                                1000

# Sort the file results by how often they have been opened
Search.SortOpenFilesByOpenFrequency:              false

# Comma separated list of paths. The parser will parse all files found in these directories. End with * to add files recursively.
Parser.AdditionalIncludePaths:                    

# Use the current word as the initial search text when opening the Search panel
Search.SearchCurrentWord:                         true

# Automatically fix . and -> depending if the variable is a pointer
Parser.AutoFixIndirection:                        false

# Automatically add a close brace after typing an open brace
Editor.AutoAddCloseBrace:                         false


# The maximum number of cores to use, -1 to use SystemCoreCount*0.8 (requires restart)
MaxCoreCount:                                     -1
```

**Comment by slynch8** [comment]
thanks. Still doesn't happen though. Just to check we're talking about the same thing, this is what happens for me:

![indent](https://user-images.githubusercontent.com/7141856/104247556-effe9f80-545f-11eb-9647-69a03593cdff.gif)

**Comment by Muzza** [comment]
Here's mine.
![Indent](https://user-images.githubusercontent.com/1583441/104248080-f3744400-54bc-11eb-9c00-b229f6af3ea6.gif)

**Comment by Muzza** [comment]
I've tried making my code identical to yours, and it still happens for me.

**Comment by slynch8** [comment]
same, feels like this is much harder to track down than it should be :-)

can you try copy-pasting this code:
```
void test( void )
{
	if( 1 )
	{
		if( 2 )

```

**Comment by Muzza** [comment]
Done, and I still get the problem. Weird one.

**Comment by slynch8** [comment]
ah, got it in 10x Notepad! Maybe only happens opening a file outside a workspace.

**Comment by Muzza** [comment]
I first experienced it editing multiple files that are all in my main workspace.
For the test, I opened a new folder workspace.

**Comment by slynch8** [comment]
I've just deployed a new build with _no changes_ and it seems to have fixed it. 

It seems like that 20.10 build only had some of my indentation changes, although I've no idea how that could have happened, because the deployment process is fully automated.

**Comment by Muzza** [comment]
Working for me now, thanks.

**Comment by Muzza** [comment]
Hate to say it, but this bug has come back with today's update.

**Comment by slynch8** [comment]
fixed! (0.20.13.0)

This time I was getting it in the debugger, so the fix was simple. Still a mystery why I wasn't getting it yesterday though.
